BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Models/SpanModel.cs

```diff
@@ -3,71 +3,123 @@
+        public SpanModel(Span span, PerformanceConfiguration config)
         {
-            if (attr.Value is string[] stringArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeStringArrayValueModel(stringArray)));
-            }
-            else if (attr.Value is long[] intArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeIntArrayValueModel(intArray)));
-            }
-            else if (attr.Value is bool[] boolArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeBoolArrayValueModel(boolArray)));
-            }
-            else if (attr.Value is double[] doubleArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeDoubleArrayValueModel(doubleArray)));
-            }
-            else if (attr.Value is string strValue)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeStringValueModel(strValue)));
-            }
-            else if (attr.Value is long intValue)
+            name = span.Name;
+            kind = (int)span.Kind;
+            spanId = span.SpanId;
+            traceId = span.TraceId.Replace("-", string.Empty);
+            parentSpanId = span.ParentSpanId;
+            startTimeUnixNano = GetNanoSeconds(span.StartTime);
+            endTimeUnixNano = GetNanoSeconds(span.EndTime);
+
+            foreach (var attr in span.GetAttributes())
             {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeIntValueModel(intValue)));
+                if (string.IsNullOrEmpty(attr.Key) || attr.Value == null)
+                {
+                    continue;
+                }
+
+                if (attr.Key.Length > 128)
+                {
+                    span.DroppedAttributesCount++;
+                    MainThreadDispatchBehaviour.Instance().LogWarning(string.Format(KEY_LIMIT_WARNING_MESSAGE, attr.Key, span.Name));
+                    continue;
+                }
+
+                if (attr.Value is string[] stringArray)
+                {
+                    var truncatedStringArray = TruncateArrayIfNeeded(stringArray, config.AttributeArrayLengthLimit, attr.Key, span.Name);
+                    var valueLengthCheckedStringArray = new string[truncatedStringArray.Length];
+                    for (int i = 0; i < truncatedStringArray.Length; i++)
+                    {
+                        var strValue = truncatedStringArray[i];
+                        if (strValue.Length > config.AttributeStringValueLimit)
+                        {
+                            int truncatedLength = strValue.Length - config.AttributeStringValueLimit;
+                            strValue = strValue.Substring(0, config.AttributeStringValueLimit) + $"*** {truncatedLength} CHARS TRUNCATED";
+                        }
+                        valueLengthCheckedStringArray[i] = strValue;
```

Comment:
Feels like this should be a `TruncateStringIfNeeded` method so it's not repeated.